src/cellfinder_core/detect/filters/plane/tile_walker.py

```diff
@@ -1,69 +1,80 @@
 import math
 
 import numpy as np
+from numba import jit
 
-from cellfinder_core.detect.filters.plane.base_tile_filter import (
-    OutOfBrainTileFilter,
-    is_low_average,
-)
 
+class TileWalker:
+    """
+    A class to segment a 2D image into tiles, and mark each of the
+    tiles as good or bad depending on whether the average image
+    value in each tile is above a threshold.
+
+    The threshold is set using the tile of data in the corner.
+    The mean and standard deviation of this tile is calculated, and
+    the threshold set at 1 + mean + (2 * stddev).
+
+    Attributes
+    ----------
+    good_tiles_mask :
+        An array whose entries correspond to each tile.
+        The values are set in self.walk_out_of_brain_only()
+    """
 
-class TileWalker(object):
     def __init__(self, img, soma_diameter):
         self.img = img
-        self.thresholded_img = img.copy()
         self.img_width, self.img_height = img.shape
-        self.soma_diameter = soma_diameter
-        self.tile_width = self.soma_diameter * 2
-        self.tile_height = self.soma_diameter * 2
+        self.tile_width = soma_diameter * 2
+        self.tile_height = soma_diameter * 2
 
         n_tiles_width = math.ceil(self.img_width / self.tile_width)
         n_tiles_height = math.ceil(self.img_height / self.tile_height)
         self.good_tiles_mask = np.zeros(
             (n_tiles_width, n_tiles_height), dtype=bool
         )
 
-        self.x = 0
-        self.y = 0
-        self.tile_idx = 0
-
-        corner_intensity = img[
-            0 : self.tile_width, 0 : self.tile_height
-        ].mean()
-        corner_sd = img[0 : self.tile_width, 0 : self.tile_height].std()
-        out_of_brain_threshold = (
-            corner_intensity + (2 * corner_sd)
-        ) + 1  # add 1 to ensure not 0, as disables
-
-        self.ftf = OutOfBrainTileFilter(
-            out_of_brain_intensity_threshold=out_of_brain_threshold
-        )
+        corner_tile = img[0 : self.tile_width, 0 : self.tile_height]
+        corner_intensity = np.mean(corner_tile)
+        corner_sd = np.std(corner_tile)
+        # add 1 to ensure not 0, as disables
+        self.out_of_brain_threshold = (corner_intensity + (2 * corner_sd)) + 1
 
     def _get_tiles(self):  # WARNING: crops to integer steps
```

Comment:
should the `# WARNING: ` become part of the docstring?

Reply:
I've taken a guess at what this warning means, and migrated it into the docstring